packages/ai/core/tool/mcp/mcp-client.ts

```diff
@@ -78,10 +78,16 @@ class MCPClient {
     onUncaughtError,
   }: MCPClientConfig) {
     this.onUncaughtError = onUncaughtError;
-    this.transport = createMcpTransport(transportConfig);
-    this.transport.onClose = () => this.onClose();
-    this.transport.onError = (error: Error) => this.onError(error);
-    this.transport.onMessage = message => {
+
+    if ('type' in transportConfig) {
```

Comment:
wonder if this is good enough of a check - bc the `type` property is very common.

we could also turn it around and check for a few function properties of `MCPTransport`